nifi-api/src/main/java/org/apache/nifi/controller/status/PerformanceMetrics.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.controller.status;
+
+public class PerformanceMetrics implements Cloneable {
+
+    private String identifier;
+    private long cpuTime;
+    private long cpuTimePercentage;
+    private long readTime;
+    private long readTimePercentage;
```

Comment:
Calculating all of these percent values could become expensive for flows thousands of Processors. In the interest of making this information available it seems like it would be better to avoid this percentage calculation entirely.